src/main/fc/rc_modes.c

```diff
@@ -134,13 +134,7 @@ void rcModeUpdate(boxBitmask_t *newState)
 
 bool isModeActivationConditionPresent(boxId_e modeId)
 {
-    for (int index = 0; index < MAX_MODE_ACTIVATION_CONDITION_COUNT; index++) {
-        if (modeActivationConditions(index)->modeId == modeId && IS_RANGE_USABLE(&modeActivationConditions(index)->range)) {
-            return true;
-        }
-    }
-
-    return false;
+    return specifiedConditionCountPerMode[modeId] > 0;
```

Comment:
Nice find! I forgot that we actually do pre-calculate this :+1:

Reply:
I made a system that used a bitflag and as I was searching for places that should trigger it updating, I found that this does the same thing :)